src/main/java/Parser.java

```diff
@@ -0,0 +1,121 @@
+import java.util.LinkedList;
+
+public class Parser {
+    
+    public static LinkedList<String> parseAndExecute(String input, TaskList tasks, FileHandler fileHandler) 
+            throws DukeException {
+        //Split input incase it has a command and argument
+        String[] inputArray = input.split(" ", 2);
+
+        String[] cmdArgs = new String[0];
+        Task newTask;
+
+        LinkedList<String> result = new LinkedList<>();
+
+        switch (inputArray[0]) { 
+        case "list":
+            Task currTask;
+
+            //Append to result
+            for (int i = 1; i <= tasks.getTaskListLength(); i++) {
+                currTask = tasks.getTask(i - 1);
+                result.add(i + ". " + currTask);
+            }
+
+            break;
+        case "done":

```

Comment:
Same as above. What I did was to generate a DoneCommand and let it take care of all the following lines in `public void execute()` method. You can read more about it by searching Command pattern, one of the software engineering patterns (in CS3219 I think).